cmds/core/uptime/uptime_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"testing"
+	"time"
+)
+
+type UptimeOutput struct {
+	uptime time.Time
+	err    string
+}
+
+type LoadAvgOutput struct {
+	loadavg string
+	err     string
+}
+
+func TestUptimeGoodData(t *testing.T) {
+	time := time.Date(0001, 1, 15, 5, 35, 49, 0, time.UTC)
+	want := UptimeOutput{time, ""}
+	uptime, err := uptime("1229749 1422244")
+	var got UptimeOutput
+	if err == nil {
+		got = UptimeOutput{*uptime, ""}
+	} else {
+		got = UptimeOutput{*uptime, err.Error()}
+	}
+	if !got.uptime.Equal(want.uptime) {
+		t.Errorf("the uptime got is not correct:want %v , got %v", want.uptime, got.uptime)
+	}
+	if got.err != want.err {
+		t.Errorf("expected no error from a good proc/uptime file but got this error:%v", got.err)
+	}
+}
+
+func TestUptimeBadData(t *testing.T) {
+	time := time.Date(0000, 0, 0, 0, 0, 0, 0, time.UTC)
+	_, err := uptime("string")
+	var got UptimeOutput
+	if err == nil {
+		got = UptimeOutput{time, ""}
```

Comment:
You shouldn't fake up a time. Actually take the time output from calling uptime. Check that you get nil. uptime should return a nil Time pointer, so check that you actually get nil back.

Reply:
This is also why I think it's not useful to have a the UptimeOutput struct. You can change the UptimeOutput struct to hold a time pointer instead